mixin/package.go

```diff
@@ -110,27 +110,32 @@ func addPackage(pkg string, build bool) (string, error) {
 	b.NumBundleWorkers = runtime.NumCPU()
 	b.NumFullfileWorkers = runtime.NumCPU()
 
-	bundle, err = getPackageRepo(pkg, ver, b.Config.Builder.DNFConf)
+	rpms, err := helpers.ListVisibleFiles(b.Config.Mixer.LocalRPMDir)
 	if err != nil {
 		return "", err
 	}
 
-	err = b.EditBundles([]string{bundle}, true, true, false)
-	if err != nil {
-		return "", err
+	if len(rpms) > 0 {
+		// initialize local repo automatically
+		runInitRepo(&cobra.Command{}, []string{})
+
+		err = b.AddRPMList(rpms)
+		if err != nil {
+			return "", err
+		}
 	}
 
-	err = appendToFile(filepath.Join(mixWS, "local-bundles", bundle), fmt.Sprintf("%s\n", pkg))
+	bundle, err = getPackageRepo(pkg, ver, b.Config.Builder.DNFConf)
 	if err != nil {
 		return "", err
 	}
 
-	rpms, err := helpers.ListVisibleFiles(b.Config.Mixer.LocalRPMDir)
+	err = b.EditBundles([]string{bundle}, true, true, false)
```

Comment:
How is anyone expected to figure out this api just from examples?

Perhaps instead of the 3 bool flags it could take a struct with 3 bool flags so at least they get names?

Perhaps have a helper routine "EditBundlesWithoutEditor" which passes true as the first flag?

Go the whole https://commandcenter.blogspot.com.au/2014/01/self-referential-functions-and-design.html https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis route?

Reply:
I agree that the api for that sucks, but since I just moved the call and didn't change it I'd rather not do the legwork of fixing the API in this PR.